plugins/modules/foreman_compute_resource.py

```diff
@@ -226,14 +220,23 @@ def main():
         if 'locations' in entity_dict:
             entity_dict['locations'] = module.find_resources_by_title('locations', entity_dict['locations'], thin=True)
 
+        if 'provider' in entity_dict:
+            entity_dict['provider'] = entity_dict['provider'].title()
```

Comment:
I know this was in the old code too. But I don't see the reason for it. The API will `.downcase` it anyways: https://github.com/theforeman/foreman/blob/d84dfbbd486d98d3cc1160dab13dd7ebf6686460/app/models/compute_resource.rb#L89-L96

Reply:
and the "real" names are different anyways: https://github.com/theforeman/foreman/blob/develop/app/models/compute_resource.rb#L45-L55

Reply:
I fear we need something like this, because foreman does not accept the downcased version.

Reply:
interesting. thanks for testing. then let's keep it.

Reply:
Have a look again: I reintroduced the provider_info thing, because title would not work for CamelCase.

Reply:
Not that we have CamelCase (today), but you're right, let's be explicit.